backend/chat/chatHandlers.js

```
@@ -0,0 +1,135 @@
+module.exports = (io, socket) => {
+  const Joi = require("joi");
+
+  const chatDb = require("../models/chat");
+
+  const sendMessage = async (payload, callback) => {
+    if (callback && typeof callback !== "function") {
+      socket.disconnect();
+    }
+
+    if (!socket.data.chatId) {
+      callback({
+        status: "No room",
+        message: "You must join a room before sending a message"
+      });
+      socket.disconnect();
+    }
+
+    const schema = Joi.object({
+      name: Joi.string().required(),
+      message: Joi.string().required()
+    });
+
+    const { error } = schema.validate(payload);
+    if (error) {
+      return callback({ status: "Bad request", message: error.message });
+    }
+
+    io.to(socket.data.chatId).emit("message", payload);
+
+    try {
+      const success = await chatDb.addMessage(
+        socket.data.chatId,
+        socket.data.user.id,
+        payload.message
+      );
+      if (!success) throw new Error("Failed to save message");
+    } catch (error) {
+      callback("Failed to save message");
+    }
+  };
+
+  const joinDirectChat = async (friendUser, callback) => {
+    if (typeof callback !== "function") {
+      return socket.disconnect();
+    }
+
+    const schema = Joi.object({
+      id: Joi.string().uuid().required(),
+      firstname: Joi.string().required(),
+      lastname: Joi.string().required()
+    });
+
+    const { error } = schema.validate(friendUser);
+    if (error) {
+      return callback({ status: "Bad request", message: error.message });
+    }
+
+    const userId = socket.data.user?.id;
+    const usersAreFriends = await chatDb.checkFriends(userId, friendUser.id);
+    if (usersAreFriends) {
+      let chatId = await chatDb.getDirectChatID(userId, friendUser.id);
```

DB could throw error. Errors are not being handled for these two db functions

---

Resolved by adding TODO comment pointing it out.